### ISSUE TITLE
uploading iso -> 500 Internal Server Erro

### ISSUE THREAD

**Issue by nORKy54** [hashicorp/packer-plugin-vsphere#267]
#### Overview of the Issue

Packer create an iso (because I use a cd_content directive) and try to upload to my datastore. But it failed with "errored after 447 milliseconds 412 microseconds: 500 Internal Server Error"

#### Reproduction Steps

Configure a source and a build with a cd_content and run packer

### Packer Version

1.8.6

### Plugin Version and Builders

How find version ?? Log says "packer-builder-vsphere-clone plugin: Plugin address: unix /tmp/packer-plugin3064546192" but tmp files are deleted and the end of the run. So, is is the latest downloaded each time ?

- [] `vsphere-iso`
- [x] `vsphere-clone`

### VMware vSphere Version

- 8.0

### Guest Operating System

Clone from a CentOS9 Stream

### Simplified Packer Buildfile

```
build {
    name = "mySubuild"

    source "source.vsphere-clone.centos9-stream" {
        vm_name  = "${local.vm_name}-mySubuil"
        cd_content       = {
            "README.md"  =  "Adding a CDrom..."
        }
    }
}
````

### Operating System and Environment Details

Rocky Linux with kernel 5.14

### Log Fragments and `crash.log` Files
<details>
  <summary>TRACE</summary>
  
```
2023/03/29 18:31:26 [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/03/29 18:31:26 [TRACE] discovering plugins in /home/myuser/.local/bin
2023/03/29 18:31:26 [TRACE] discovering plugins in /home/myuser/.config/packer/plugins
2023/03/29 18:31:26 [DEBUG] Discovered plugin: windows-update = /home/myuser/.config/packer/plugins/github.com/rgl/windows-update/packer-plugin-windows-update_v0.14.1_x5.0_linux_amd64
2023/03/29 18:31:26 found external [-packer-default-plugin-name-] provisioner from windows-update plugin
2023/03/29 18:31:26 [TRACE] discovering plugins in .
2023/03/29 18:31:26 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/03/29 18:31:26 [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/myuser/.packerconfig
2023/03/29 18:31:26 [WARN] Config file doesn't exist: /home/myuser/.packerconfig
2023/03/29 18:31:26 [INFO] Setting cache directory: /home/myuser/.cache/packer
2023/03/29 18:31:26 [TRACE] init: plugingetter.ListInstallationsOptions{FromFolders:[]string{"/home/myuser/.local/bin/packer", ".", "/home/myuser/.config/packer/plugins"}, BinaryInstallationOptions:plugingetter.BinaryInstallationOptions{APIVersionMajor:"5", APIVersionMinor:"0", OS:"linux", ARCH:"amd64", Ext:"", Checksummers:[]plugingetter.Checksummer{plugingetter.Checksummer{Type:"sha256", Hash:(*sha256.digest)(0xc0000f0400)}}}}
2023/03/29 18:31:26 [INFO] (telemetry) Finalizing.
2023/03/29 18:31:26 waiting for all plugin processes to complete...
2023/03/29 18:31:26 [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/03/29 18:31:26 [TRACE] discovering plugins in /home/myuser/.local/bin
2023/03/29 18:31:26 [TRACE] discovering plugins in /home/myuser/.config/packer/plugins
2023/03/29 18:31:26 [DEBUG] Discovered plugin: windows-update = /home/myuser/.config/packer/plugins/github.com/rgl/windows-update/packer-plugin-windows-update_v0.14.1_x5.0_linux_amd64
2023/03/29 18:31:26 found external [-packer-default-plugin-name-] provisioner from windows-update plugin
2023/03/29 18:31:26 [TRACE] discovering plugins in .
2023/03/29 18:31:26 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/03/29 18:31:26 [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/myuser/.packerconfig
2023/03/29 18:31:26 [WARN] Config file doesn't exist: /home/myuser/.packerconfig
2023/03/29 18:31:26 [INFO] Setting cache directory: /home/myuser/.cache/packer
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_network, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_folder, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_guest_os_keyboard, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_guest_os_name, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_cdrom_type, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for build_password, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for build_password_encrypted, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_vm_version, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_http_port_max, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_boot_wait, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_shutdown_timeout, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for iso_checksum, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_datacenter, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_username, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_network_card, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_content_library_name, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_disk_thin_provisioned, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_content_library_ovf, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_ip_wait_timeout, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for build_organization, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_cluster, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_guest_os_language, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_guest_os_timezone, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_disk_controller_type, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for communicator_timeout, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for scripts-base, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for iso_url, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_boot_order, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_vcenter, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_cpu_sockets, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_cpu_cores, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for inline-base, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_guest_os_type, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_http_port_min, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for build_username, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_insecure, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_guest_os_version, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_mem_size, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_content_library_destroy, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_guest_os_family, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_firmware, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_template_conversion, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_datastore, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_disk_size, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_content_library_skip_export, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_ovf_export_overwrite, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_mem_hot_add, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for communicator_port, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_remove_cdrom, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_ovf_export_enabled, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vsphere_password_vpath, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for vm_cpu_hot_add, so skipping
2023/03/29 18:31:26 [TRACE] validateValue: not active for common_tools_upgrade_policy, so skipping
2023/03/29 18:31:26 [TRACE] Starting internal plugin packer-builder-vsphere-clone
2023/03/29 18:31:26 Starting plugin: /home/myuser/.local/bin/packer []string{"/home/myuser/.local/bin/packer", "plugin", "packer-builder-vsphere-clone"}
2023/03/29 18:31:26 Waiting for RPC address for: /home/myuser/.local/bin/packer
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/myuser/.packerconfig
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: [WARN] Config file doesn't exist: /home/myuser/.packerconfig
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: [INFO] Setting cache directory: /home/myuser/.cache/packer
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: args: []string{"packer-builder-vsphere-clone"}
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: Plugin address: unix /tmp/packer-plugin1030611791
2023/03/29 18:31:26 Received unix RPC address for /home/myuser/.local/bin/packer: addr is /tmp/packer-plugin1030611791
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: Waiting for connection...
2023/03/29 18:31:26 packer-builder-vsphere-clone plugin: Serving a plugin connection...
2023/03/29 18:31:26 [TRACE] Starting internal plugin packer-provisioner-ansible
2023/03/29 18:31:26 Starting plugin: /home/myuser/.local/bin/packer []string{"/home/myuser/.local/bin/packer", "plugin", "packer-provisioner-ansible"}
2023/03/29 18:31:26 Waiting for RPC address for: /home/myuser/.local/bin/packer
2023/03/29 18:31:26 packer-provisioner-ansible plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/03/29 18:31:26 packer-provisioner-ansible plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/03/29 18:31:26 packer-provisioner-ansible plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/myuser/.packerconfig
2023/03/29 18:31:26 packer-provisioner-ansible plugin: [WARN] Config file doesn't exist: /home/myuser/.packerconfig
2023/03/29 18:31:26 packer-provisioner-ansible plugin: [INFO] Setting cache directory: /home/myuser/.cache/packer
2023/03/29 18:31:26 packer-provisioner-ansible plugin: args: []string{"packer-provisioner-ansible"}
2023/03/29 18:31:26 packer-provisioner-ansible plugin: Plugin address: unix /tmp/packer-plugin3521502175
2023/03/29 18:31:26 packer-provisioner-ansible plugin: Waiting for connection...
2023/03/29 18:31:26 Received unix RPC address for /home/myuser/.local/bin/packer: addr is /tmp/packer-plugin3521502175
2023/03/29 18:31:26 packer-provisioner-ansible plugin: Serving a plugin connection...
2023/03/29 18:31:27 packer-provisioner-ansible plugin: ansible-playbook version: 2.13.6
2023/03/29 18:31:27 [TRACE] Starting internal plugin packer-provisioner-file
2023/03/29 18:31:27 Starting plugin: /home/myuser/.local/bin/packer []string{"/home/myuser/.local/bin/packer", "plugin", "packer-provisioner-file"}
2023/03/29 18:31:27 Waiting for RPC address for: /home/myuser/.local/bin/packer
2023/03/29 18:31:27 packer-provisioner-file plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/03/29 18:31:27 packer-provisioner-file plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/03/29 18:31:27 packer-provisioner-file plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/myuser/.packerconfig
2023/03/29 18:31:27 packer-provisioner-file plugin: [WARN] Config file doesn't exist: /home/myuser/.packerconfig
2023/03/29 18:31:27 packer-provisioner-file plugin: [INFO] Setting cache directory: /home/myuser/.cache/packer
2023/03/29 18:31:27 packer-provisioner-file plugin: args: []string{"packer-provisioner-file"}
2023/03/29 18:31:27 packer-provisioner-file plugin: Plugin address: unix /tmp/packer-plugin2784358383
2023/03/29 18:31:27 Received unix RPC address for /home/myuser/.local/bin/packer: addr is /tmp/packer-plugin2784358383
2023/03/29 18:31:27 packer-provisioner-file plugin: Waiting for connection...
2023/03/29 18:31:27 packer-provisioner-file plugin: Serving a plugin connection...
2023/03/29 18:31:27 [TRACE] Starting internal plugin packer-provisioner-shell
2023/03/29 18:31:27 Starting plugin: /home/myuser/.local/bin/packer []string{"/home/myuser/.local/bin/packer", "plugin", "packer-provisioner-shell"}
2023/03/29 18:31:27 Waiting for RPC address for: /home/myuser/.local/bin/packer
2023/03/29 18:31:27 packer-provisioner-shell plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/03/29 18:31:27 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/03/29 18:31:27 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /home/myuser/.packerconfig
2023/03/29 18:31:27 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/myuser/.packerconfig
2023/03/29 18:31:27 packer-provisioner-shell plugin: [INFO] Setting cache directory: /home/myuser/.cache/packer
2023/03/29 18:31:27 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2023/03/29 18:31:27 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin474121248
2023/03/29 18:31:27 packer-provisioner-shell plugin: Waiting for connection...
2023/03/29 18:31:27 Received unix RPC address for /home/myuser/.local/bin/packer: addr is /tmp/packer-plugin474121248
2023/03/29 18:31:27 packer-provisioner-shell plugin: Serving a plugin connection...
mybuild.vsphere-clone.centos9-stream: output will be in this color.
2023/03/29 18:31:27 Build debug mode: false
2023/03/29 18:31:27 Force build: true
2023/03/29 18:31:27 On error:
2023/03/29 18:31:27 Waiting on builds to complete...
2023/03/29 18:31:27 Starting build run: mybuild.vsphere-clone.centos9-stream
2023/03/29 18:31:27 Running builder:
2023/03/29 18:31:27 [INFO] (telemetry) Starting builder vsphere-clone.centos9-stream

==> mybuild.vsphere-clone.centos9-stream: Creating CD disk...
2023/03/29 18:31:27 packer-builder-vsphere-clone plugin: CD path: /tmp/packer477915425.iso
2023/03/29 18:31:27 packer-builder-vsphere-clone plugin: Executing: /usr/bin/xorriso [-as genisoimage -rock -joliet -volid packer -output /tmp/packer477915425.iso /tmp/packer_to_cdrom2287267623]
    mybuild.vsphere-clone.centos9-stream: xorriso 1.5.4 : RockRidge filesystem manipulator, libburnia project.
    mybuild.vsphere-clone.centos9-stream: Drive current: -outdev 'stdio:/tmp/packer477915425.iso'
    mybuild.vsphere-clone.centos9-stream: Media current: stdio file, overwriteable
    mybuild.vsphere-clone.centos9-stream: Media status : is blank
    mybuild.vsphere-clone.centos9-stream: Media summary: 0 sessions, 0 data blocks, 0 data, 1827m free
    mybuild.vsphere-clone.centos9-stream: xorriso : WARNING : -volid text does not comply to ISO 9660 / ECMA 119 rules
    mybuild.vsphere-clone.centos9-stream: Added to ISO image: directory '/'='/tmp/packer_to_cdrom2287267623'
    mybuild.vsphere-clone.centos9-stream: xorriso : UPDATE :       1 files added in 1 seconds
    mybuild.vsphere-clone.centos9-stream: xorriso : UPDATE :       1 files added in 1 seconds
    mybuild.vsphere-clone.centos9-stream: ISO image produced: 184 sectors
    mybuild.vsphere-clone.centos9-stream: Written to medium : 184 sectors at LBA 0
    mybuild.vsphere-clone.centos9-stream: Writing to 'stdio:/tmp/packer477915425.iso' completed successfully.
    mybuild.vsphere-clone.centos9-stream: Done copying paths from CD_dirs
==> mybuild.vsphere-clone.centos9-stream: Uploading packer477915425.iso to packer_cache/packer477915425.iso
2023/03/29 18:31:27 packer-builder-vsphere-clone plugin: Deleting CD disk: /tmp/packer477915425.iso
2023/03/29 18:31:27 [INFO] (telemetry) ending vsphere-clone.centos9-stream
==> Wait completed after 343 milliseconds 890 microseconds
2023/03/29 18:31:27 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2023/03/29 18:31:27 machine readable: mybuild.vsphere-clone.centos9-stream,error []string{"500 Internal Server Error"}
Build 'mybuild.vsphere-clone.centos9-stream' errored after 343 milliseconds 666 microseconds: 500 Internal Server Error
==> Builds finished but no artifacts were created.

2023/03/29 18:31:27 [INFO] (telemetry) Finalizing.
==> Wait completed after 343 milliseconds 890 microseconds

==> Some builds didn't complete successfully and had errors:
--> mybuild.vsphere-clone.centos9-stream: 500 Internal Server Error
```
</details>

**Comment by tenthirtyam** [comment]
> How find version ?? Log says "packer-builder-vsphere-clone plugin: Plugin address: unix /tmp/packer-plugin3064546192" but tmp files are deleted and the end of the run. So, is is the latest downloaded each time ?

Example command:

```console
> packer plugins installed
/Users/johnsonryan/.packer.d/plugins/github.com/rgl/windows-update/packer-plugin-windows-update_v0.14.1_x5.0_darwin_amd64
/Users/johnsonryan/.packer.d/plugins/github.com/ethanmdavidson/git/packer-plugin-git_v0.3.4_x5.0_darwin_amd64
/Users/johnsonryan/.packer.d/plugins/github.com/hashicorp/virtualbox/packer-plugin-virtualbox_v1.0.4_x5.0_darwin_amd64
/Users/johnsonryan/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.1_x5.0_darwin_amd64
```

**Comment by nORKy54** [comment]
I'm sorry, I found the error (thanks vpxd.log). I need the "Cryptographer.Access" permission.
Sorry for this case.

**Comment by tenthirtyam** [comment]
> I'm sorry, I found the error (thanks vpxd.log). I need the "Cryptographer.Access" permission.

Acknowledged.

**Comment by natedogith1** [comment]
Got the same issue, couldn't check vpxd.log, but adding Cryptographer.Access fixed the issue.  The documentation doesn't mention this permission.  Can this be re-opened to fix the documentation, or should a new issue be created for that?
